### PR TITLE
New algorithm

### PR DESCRIPTION
**Pull request by xeodou**
- New Algorithm
- Add simple CLI UI

image

### CODE REVIEW

File: raffle.js [hunk 1]
```diff
@@ -1,41 +1,158 @@
 #!/usr/bin/env node
```
Comment (fraserxu) [inline raffle.js:1]: If you are using `const`, you probably need to add the least required version of node to readme file?

Comment (xeodou) [reply raffle.js:1]: will add `engine` property in `package.json`

**Comment by fraserxu** [comment]
好高端。

File: raffle.js [hunk 2]
```diff
@@ -1,41 +1,158 @@
...
+    .sort((a, b) => b.weight - a.weight)
+    .map(l => {
+      delete l.weight
+      return l
+    })
+    .slice(0, length || 1)
+}
+
+// Probability Graph
+const line = grid.set(0, 2, 1, 2, contrib.line, {
```
Comment (fraserxu) [inline raffle.js:87]: Maybe you can separate the function to generate number and draw graph into different files so it's easy to read the code and review?

Comment (xeodou) [reply raffle.js:87]: 😢

**Comment by xeodou** [comment]
先合并了。

**Pull request merged**